### PR TITLE
EWPP-376: Add Contact Messages sanitization.

### PR DESCRIPTION
**Pull request by georgebaev**
@drishu This adds the Contact Form Messages sanitisation.

### CODE REVIEW

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  /**
+   * Sets the output message.
+   *
+   * @hook on-event sql-sanitize-confirms
+   *
+   * @inheritdoc
+   */
+  public function messages(&$messages, InputInterface $input) {
```
[the remark this answers is not in the document]
Comment (georgebaev) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php]: @drishu I added the parameter type hint.

[the remark this answers is not in the document]
Comment (drishu) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php:88]: actually I saw that SanitizePluginInterface needs to be implemented

**Comment by georgebaev** [comment]
@drishu I altered the phpunit test and modified the call to the sanitize() method.

**Review by drishu — DISMISSED** [review]

File: tests/src/Kernel/SanitizeContactFormFieldsTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      'ip_address' => '8.8.8.8',
+      'subject' => 'Test subject',
+      'message' => 'Test message',
+      'oe_country_residence' => 'http://publications.europa.eu/resource/authority/country/BEL',
+      'oe_telephone' => '0123456',
+      'oe_topic' => 'Topic name',
+    ];
+
+    $message = Message::create($data);
+
```
Comment (sergepavle) [inline tests/src/Kernel/SanitizeContactFormFieldsTest.php:45]: not necessary empty line.

File: tests/src/Kernel/SanitizeContactFormFieldsTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    $message->save();
+
+    $message_id = $message->id();
+
```
Comment (sergepavle) [inline tests/src/Kernel/SanitizeContactFormFieldsTest.php:49]: not necessary empty line.

File: oe_contact_forms.services.yml [hunk 4]
```diff
@@ -7,3 +7,8 @@ services:
     class: Drupal\oe_contact_forms\Access\ContactFormsAccessCheck
     tags:
       - { name: access_check, applies_to: _oe_contact_forms_access_check }
+  oe_contact_forms.contact.commands:
```
Comment (sergepavle) [inline oe_contact_forms.services.yml:10]: The name of the service can be improved. At least it should reflect the purpose of service.

**Review by sergepavle — CHANGES_REQUESTED** [review]
I have left a few minor comments. Everything else seems to work well.
Did manual testing. Command works correctly.

**Comment by georgebaev** [comment]
@sergepavle thank you for the review! I implemented the requested changes.

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      ->expression('message', "CONCAT('message-by-', id)")
+      ->expression('oe_country_residence', "CONCAT('residence-in-', id)")
+      ->expression('oe_telephone', "CONCAT('+000-', id)")
+      ->expression('oe_topic', "CONCAT('topic-', id)")
+      ->fields([
+        'ip_address' => '127.0.0.1',
+      ])
+      ->execute();
+
+    $this->logger->notice('Contact messages data has been sanitized.');
```
Comment (sergepavle) [inline src/Commands/sql/SanitizeContactFormFieldsCommands.php:78]: I am sorry @georgebaev . Yesterday by mistake have removed more correct comment. It shouldn't be 'notice' but 'success':
```suggestion
    $this->logger->success('Contact messages data sanitized.');
```

**Review by sergepavle — COMMENTED** [review]
Additional comment

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php (hunk 5 above)
Comment (georgebaev) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php]: @sergepavle this is fixed.

Comment (drishu) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php]: @sergepavle , drone fails, and if we look at https://github.com/php-fig/log/blob/master/Psr/Log/LoggerInterface.php there is not success method.
We could use info().

Comment (georgebaev) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php]: The `info()` method does not display any message in the `$ drush sql-sanitize` command result log. I reverted back the logger message type to `notice`.

Comment (sergepavle) [reply src/Commands/sql/SanitizeContactFormFieldsCommands.php]: I just suggested make messaging more consistent with other sanitization commands where used logger:
\Drush\Drupal\Commands\sql\SanitizeSessionsCommands::sanitize

**Review by sergepavle — APPROVED** [review]

**Comment by hernani** [comment]
One point:
- The sanitization is happening on all forms. As the component is only handling corporate forms we should only handle the sanitization also in corporate forms and not affect other forms that are not controlled by this module. Please sanitize only corporate form and add test to check non-corporate form is not sanitized.

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      ->expression('mail', "CONCAT('user+', id, '@example.com')")
+      ->expression('subject', "CONCAT('subject-by-', id)")
+      ->expression('message', "CONCAT('message-by-', id)")
+      ->expression('oe_country_residence', "CONCAT('residence-in-', id)")
+      ->expression('oe_telephone', "CONCAT('+000-', id)")
+      ->expression('oe_topic', "CONCAT('topic-', id)")
+      ->fields([
+        'ip_address' => '127.0.0.1',
+      ])
+      ->execute();
```
Comment (hernani) [inline src/Commands/sql/SanitizeContactFormFieldsCommands.php:76]: This does not handle any sort of cache clear.

**Review by hernani — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+use Symfony\Component\Console\Input\InputInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
+
+/**
+ * Sanitizes the contact forms related data.
+ */
+class SanitizeContactFormFieldsCommands extends DrushCommands implements SanitizePluginInterface {
```
Comment (upchuk) [inline src/Commands/sql/SanitizeContactFormFieldsCommands.php:19]: `ContactFormSanitizeCommand`

File: src/Commands/sql/SanitizeContactFormFieldsCommands.php [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+          $message->set('oe_country_residence', 'residence-in-' . $message->id());
+          $message->set('oe_telephone', '+000-' . $message->id());
+          $message->set('oe_topic', 'topic-' . $message->id());
+          $message->set('ip_address', '127.0.0.1');
+          $message->save();
+        }
+      }
+    }
+
+    $this->logger->notice('Contact messages data sanitized.');
```
Comment (upchuk) [inline src/Commands/sql/SanitizeContactFormFieldsCommands.php:97]: Should be success (check other plugins that do the same)

**Review by upchuk — CHANGES_REQUESTED** [review]

**Review by upchuk — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**